pkg/collector/annotations_test.go

```diff
@@ -0,0 +1,60 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/open-telemetry/opentelemetry-operator/api/v1alpha1"
+)
+
+func TestAnnotations(t *testing.T) {
+	// prepare
+	otelcol := v1alpha1.OpenTelemetryCollector{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "my-instance",
+			Namespace: "my-ns",
+		},
+		Spec: v1alpha1.OpenTelemetryCollectorSpec{
+			Config: "test",
+		},
+	}
+
+	// test
+	annotations := Annotations(otelcol)
+	assert.Equal(t, "true", annotations["prometheus.io/scrape"])
```

Comment:
Could you also add a test for the case where the annotation exists in the CR? If a user has specified `scrape` as false, we should honor that. I think the code is currently doing the opposite.

Reply:
Yes. My bad, fixing it